e107_handlers/e107_class.php

```diff
@@ -3627,17 +3627,38 @@ public static function minify($js,$options=array())
 
 	/**
 	 * Set or Retrieve WYSIWYG active status. (replaces constant  e_WYSIWYG)
-	 * @param null $val
+	 * @param bool $val if null, return current value, otherwise set value to registry
 	 * @return bool|mixed|void
 	 */
 	public static function wysiwyg($val=null)
 	{
-		
+		// Check the general wysiwyg setting
 		if (self::getPref('wysiwyg',false) != true)
 		{
 			return false; 	
 		}
 
+
```

Comment:
Thanks for this PR. I'm not a fan of this code here though.  Using e107::wysiwyg(false); in the forum plugin should be enough to disable it.

Reply:
That doesn't work. I tried it already. Look at the bbarea() function, where wysiwyg() is called and you see that any previous change will be overwritten.
Changing the wysiwyg(true) in bbarea to wysiwyg() is also a bad idea, because wysiwyg() is saving the last setting in the registry, which means that the bbarea will not be replaced by tinymce system wide when wysiwyg(false) is called anywhere in the code.
As a systemwide change is not what I want to achieve, I had to add some code that checks for the current plugin and to return a value without touching the global settings in the registry.